Submitted By: Ken Moffat <ken at linuxfromscratch dot org>
Date: 2006-12-29
Initial Package Version: 4.5.14
Upstream Status: applied
Origin: Jakub Jelinek
Description: Allows strace to compile against 2.6.18 kernel headers

--- strace-4.5.14/system.c.orig	2006-12-29 22:52:43.000000000 +0000
+++ strace-4.5.14/system.c	2006-12-29 22:53:16.000000000 +0000
@@ -27,7 +27,7 @@
  * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
  * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  *
- *	$Id: system.c,v 1.39 2006/01/12 21:21:06 roland Exp $
+ *	$Id: system.c,v 1.40 2006/12/04 13:48:10 ldv Exp $
  */
 
 #include "defs.h"
@@ -1574,6 +1574,8 @@
 #endif
 
 #ifdef LINUX
+/* Linux 2.6.18+ headers removed CTL_PROC enum.  */
+# define CTL_PROC 4
 static const struct xlat sysctl_root[] = {
 	{ CTL_KERN, "CTL_KERN" },
 	{ CTL_VM, "CTL_VM" },
@@ -1582,6 +1584,9 @@
 	{ CTL_FS, "CTL_FS" },
 	{ CTL_DEBUG, "CTL_DEBUG" },
 	{ CTL_DEV, "CTL_DEV" },
+	{ CTL_BUS, "CTL_BUS" },
+	{ CTL_ABI, "CTL_ABI" },
+	{ CTL_CPU, "CTL_CPU" },
 	{ 0, NULL }
 };
 
